agent/ecscni/plugin.go

```diff
@@ -110,92 +111,92 @@ func (client *cniClient) SetupNS(cfg *Config) (*current.Result, error) {
-func (client *cniClient) createNetworkConfig(cfg *Config, bridgeConfigFunc func(*Config) (*libcni.NetworkConfig, error)) (*libcni.NetworkConfigList, error) {
-	bridgeConfig, err := bridgeConfigFunc(cfg)
-	if err != nil {
-		return nil, err
-	}
+// add invokes the ADD command of the given plugin
+func (client *cniClient) add(runtimeConfig libcni.RuntimeConf,
+	cfg *Config,
+	pluginConfigFunc func(*Config) (string, *libcni.NetworkConfig, error)) (cnitypes.Result, error) {
 
-	eniConfig, err := client.createENINetworkConfig(cfg)
+	deviceName, networkConfig, err := pluginConfigFunc(cfg)
 	if err != nil {
 		return nil, err
 	}
+	runtimeConfig.IfName = deviceName
 
-	// We order bridgeConfig after eniConfig so that the result of the bridge
-	// plugin (which contains results of the eni plugin) can be parsed to
-	// get the IP address of the veth pair created
-	pluginConfigs := []*libcni.NetworkConfig{eniConfig, bridgeConfig}
-	networkConfigList := &libcni.NetworkConfigList{
-		CNIVersion: client.cniVersion,
-		Plugins:    pluginConfigs,
+	return client.libcni.AddNetwork(networkConfig, &runtimeConfig)
+}
+
+// del invokes the DEL command of the given plugin
+func (client *cniClient) del(runtimeConfig libcni.RuntimeConf,
+	cfg *Config,
+	pluginConfigFunc func(*Config) (string, *libcni.NetworkConfig, error)) error {
+
+	deviceName, networkConfig, err := pluginConfigFunc(cfg)
+	if err != nil {
+		return err
 	}
+	runtimeConfig.IfName = deviceName
 
-	return networkConfigList, nil
+	return client.libcni.DelNetwork(networkConfig, &runtimeConfig)
 }
 
 // createBridgeNetworkConfigWithIPAM creates the config of bridge for ADD command, where
 // bridge plugin acquires the IP and route information from IPAM
-func (client *cniClient) createBridgeNetworkConfigWithIPAM(cfg *Config) (*libcni.NetworkConfig, error) {
+func (client *cniClient) createBridgeNetworkConfigWithIPAM(cfg *Config) (string, *libcni.NetworkConfig, error) {
 	// Create the bridge config first
 	bridgeConfig := client.createBridgeConfig(cfg)
 
 	// Create the ipam config
 	ipamConfig, err := client.createIPAMConfig(cfg)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
 
 	bridgeConfig.IPAM = ipamConfig
 
-	return client.constructNetworkConfig(bridgeConfig, ECSBridgePluginName)
+	networkConfig, err := client.constructNetworkConfig(bridgeConfig, ECSBridgePluginName)
+	return defaultVethName, networkConfig, err
 }
 
 // createBridgeNetworkConfigWithoutIPAM creates the config of the bridge for removal
-func (client *cniClient) createBridgeNetworkConfigWithoutIPAM(cfg *Config) (*libcni.NetworkConfig, error) {
-	return client.constructNetworkConfig(client.createBridgeConfig(cfg), ECSBridgePluginName)
+func (client *cniClient) createBridgeNetworkConfigWithoutIPAM(cfg *Config) (string, *libcni.NetworkConfig, error) {
+	networkConfig, err := client.constructNetworkConfig(client.createBridgeConfig(cfg), ECSBridgePluginName)
+	return defaultVethName, networkConfig, err
```

Comment:
can you add `if err != nil` check here and make sure that you're wrapping the error as well?